File: src/009/p009.clj
(defn square [x] (Math/pow x 2))

(defn sqrt [x] (Math/sqrt x))

(defn sum-1000? [x] (== 1000 (reduce + 0 x)))

(defn solve []
  (let [candidates (for [b (range 700 5 -1), a (range 4 b)]
                        [a b (sqrt (+ (square a) (square b)))])]
        (transduce (comp (filter sum-1000?)
                         cat
                         (map int))
                   * 1
                   candidates)))

(prn (solve))

File: src/009/p009.py
prod = 0
for a in range(1, 1001):
    for b in range(a + 1, 1001):
        c = 1000 - a - b
        if a**2 + b**2 == c**2:
            prod = a * b * c
            break
    if prod != 0:
        break

print(prod)
